### PR TITLE
docs: add documentation on how to use GDB

### PR DESCRIPTION
**Pull request by aykevl**
This documentation isn't complete but should provide the basics that can be extended. For example:

  * How to use a different board (such as the Xiao) to debug another board.
  * Troubleshooting common issues.
  * More in-depth examples how to debug a board without built-in debug adapter.

### CODE REVIEW

**Comment by ysoldak** [comment]
macOS toolchain setup.

Install OpenOCD (https://formulae.brew.sh/formula/open-ocd)
```
brew install open-ocd
```

Install GDB
```
brew tap ARMmbed/homebrew-formulae
brew install arm-none-eabi-gcc
```
Unfortunately latest version [was not merged into official repo yet](https://github.com/ARMmbed/homebrew-formulae/pull/28) 
To install latest release (10-2020-q4-major) do this instead:
```
brew tap cesarvandevelde/homebrew-formulae
brew install arm-none-eabi-gcc
```

File: content/docs/guides/debugging.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+---
+
+Debugging using GDB can be intimidating, but it is also extremely powerful. Especially on microcontrollers, having a debugger can show you things that regular print statements cannot show you. For example:
+
+  * A debugger works even when a serial port is not available. This can happen when there is a bug in an interrupt or in initialization code before the serial port is configured.
+  * You can print a stack trace, even in a panic.
+  * Depending on the optimization level and other factors, you may be able to inspect function parameters and local variables.
+  * You can step through the code and see the results on the device as you do so.
+
+Many people are initimidated by the GDB command line, but if you're familiar with a few commands (`continue`, `breakpoint`, `next`) you can already make good use of GDB. See the [GDB tutorial]({{<ref "gdb.md">}}) for an introduction. Here we will show how to get to the GDB prompt in the first place.
```
Comment (deadprogram) [inline content/docs/guides/debugging.md]: I suggest change:

> Many people are initimidated by the GDB command line, but if you're familiar with a few commands (`continue`, `breakpoint`, `next`) you can already make good use of GDB.

to 

> By learning just a few commands such as `continue`, `breakpoint`, and `next`, you will be able to start making good use of the GDB command line.

File: content/docs/guides/debugging.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+---
+title: "Debugging"
+weight: 10
+description: |
+  Debug TinyGo programs using GDB.
+---
+
+Debugging using GDB can be intimidating, but it is also extremely powerful. Especially on microcontrollers, having a debugger can show you things that regular print statements cannot show you. For example:
```
Comment (deadprogram) [inline content/docs/guides/debugging.md]: Suggest changing

> Debugging using GDB can be intimidating, but it is also extremely powerful. Especially on microcontrollers, having a debugger can show you things that regular print statements cannot show you.

to 

>  A debugger can show you things that regular print statements cannot show you, especially on microcontrollers. The debugger used by TinyGo is called [GDB](https://www.gnu.org/software/gdb/). Debugging using GDB can sometimes be intimidating, but it is also extremely powerful.

File: content/docs/tutorials/gdb.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
+---
+title: "GDB"
+weight: 10
+description: |
+  Short introduction into debugging with GDB.
+---
+
+This tutorial assumes you already know how to enter a GDB shell, but don't know yet how to use it. Getting to a GDB shell can be easy or hard, depending on the board. For more information on how to get there, see [the debugging guide]({{<ref "debugging.md">}}).
```
Comment (deadprogram) [inline content/docs/tutorials/gdb.md]: I suggest modify this text as follows:

> The debugger used by TinyGo is called [GDB](https://www.gnu.org/software/gdb/). This tutorial assumes you already have GDB installed and know how to enter the GDB shell, but don't know how to use it yet to actually debug something. 
> For information on how to install and access GDB, see [the debugging guide]({{<ref "debugging.md">}}).

File: content/docs/tutorials/gdb.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+```
+$ tinygo gdb -target=microbit examples/microbit-blink
+[...]
+target halted due to debug-request, current mode: Thread 
+xPSR: 0xc1000000 pc: 0x0000071c msp: 0x20000800
+(gdb) 
+```
+
+The following line is relevant:
```
Comment (deadprogram) [inline content/docs/tutorials/gdb.md]: Suggest changing to

> The following line is important:

**Comment by aykevl** [comment]
I have updated the PR to include the suggested changes.

> Unfortunately latest version [was not merged into official repo yet](https://github.com/ARMmbed/homebrew-formulae/pull/28)

That's not a problem: an older GDB version will work just fine. I'm using version 8 on Linux.

**Comment by deadprogram** [comment]
Thanks @aykevl and also @ysoldak now merging.

**Pull request merged**